pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -49,15 +52,20 @@ func (d *DualWriterMode1) Create(ctx context.Context, original runtime.Object, c
 	}
 	d.recordLegacyDuration(false, mode1Str, options.Kind, method, startLegacy)
 
+	d.pendingActions.Add(1)
 	go func() {
+		defer d.pendingActions.Done()
+		ctx := context.WithoutCancel(ctx)
 		ctx, cancel := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage create timeout"))
+		defer cancel()
+		created := created.DeepCopyObject()
 		createdLegacy, err := enrichLegacyObject(original, created, true)
 		if err != nil {
-			cancel()
+			log.Error(err, "could not enrich object")
+			return
```

Comment:
`cancel` by itself only cancels the context. It does not return. Here we were not returning, hence we were still calling `d.Storage.Create` even if `enrichLegacyObject` failed.